content/2015.10.01.outreachy.rst

```diff
@@ -0,0 +1,121 @@
+  start otherwise.
+
+  We are now expanding the program to people from more groups underrepresented
+  in FOSS. We hope this effort will help many people learn how exciting, varied
+  and valuable work on FOSS projects can be and how inclusive the community
+  really is.
+
+  -- `The GNOME outreachy program wiki <https://wiki.gnome.org/Outreachy>`_
+
+In our team, and at Mozilla in general, we thrive to build an inclusive
+community. And we believe being more inclusive starts by reaching to the people
+who are often underrepresented in tech.
+
+That's why we are participating in the GNOME Outreachy program this year by
+proposing an internship around Kinto and making instances discoverable.
+
+Check out the eligibility details at the end of the article if you're
+interested!
+
+Making Kinto instances discoverable
+===================================
+
+The `Kinto project <https://kinto.readthedocs.org>`_ aims to bring storage
+instances to everyone, attached to their
+`Firefox Accounts <https://accounts.firefox.com>`_ (Well, it actually supports
+multiple authentication schemes, but we want to promote Firefox Accounts as a
+first step).
+
+Currently, Kinto is thought as a centralized server: there is one instance, and
+everyone authenticates on this instance. Items are shared between users of a
+same instance.
+
+This doesn't resonates well with multiple goals we have: scalability is harder
+when there is one endpoint, and it's also not interoperable.
+
+For instance, imagine Alice and Bob. Bob is using Mozilla's servers to store
+his data, whereas Alice deployed her own Kinto instance.
+
+There are different use cases:
+
+* Alice wants to use `Routina <https://github.com/leplatrem/routina>`_, an
+  application that stores its data inside a Kinto instance. As such, Routina
+  needs a way to discover where it should store its data, and send the requests
+  to this server;
+* Bob and Alice want to collaborate on a set of data (think about a webapp to
+  manage their shared expenses). There should be a way for Alice to host
+  everything and grant access to Bob on her data. The webapp should be able to
+  use the correct server.
+
+Here are the different steps that could allow these scenarios:
+
+* At the moment they authenticate, the client detects the email address used,
+  and relies on the domain part to do a
+  `Web Finger <https://tools.ietf.org/html/rfc7033>`_ request on the domain
+  [#]_ and for the specified user.
+* In case the identified server doesn't support WebFinger, it uses a central
+  repository to lookup where the Kinto server is located.
+* Once the Kinto server located, all requests should be issued against this
+  server.
+
+In terms of code changes, here is what it looks like (rough step by step). The
+details are still to be fleshed out, this is a rough first version:
+
+* Update the `Kinto.js client <https://github.com/Kinto/kinto.js>`_ to find the
+  server location. It should first rely on WebFinger;
```

Comment:
These bullet points should be terminated with periods, not semi-colons.

Lists *can* terminate with semi-colons if each element is a logical continuation of the leading phrase, for example:

> The colour of the ball could be red, or it could be:
>
* blue, assuming the dog is also blue;
* orange, but only if up is down;
* black (for certain values of black).

Generally speaking, however, list elements are self-contained and should therefore not be constructed this way.